### DISCUSSION TITLE
Re-enable ability to run distinct queries for Taxon

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5757] [Feature Requests]
See https://github.com/specify/specify7/pull/1019#issuecomment-973525594

Need to fix the back-end bug and then re-enable it on the front-end

**Comment by grantfitzsimmons** [comment]
@maxpatiiuk Is this fixed yet?

**Comment by maxpatiiuk** [comment]
I don't think so, but you could ask @realVinayak who might know more about this (Jason might too)

**Comment by grantfitzsimmons** [comment]
@realVinayak @melton-jason Do we have any plan to fix this? I believe this is not working still.

**Comment by ghost** [comment]
As of 9b3aa088 distinct queries on trees seem to be working for me. Just removed the condition around the checkbox, tested with taxon and geography on a fresh database. No errors and results change. Clicking on results seems functional. Perhaps something else accidentally fixed it?

**Reply by grantfitzsimmons** [reply]
Are you querying directly on the Taxon/Geography/Storage tree?

https://sp7demofish.specifycloud.org/specify/query/new/taxon/

This is the behavior I am observing in `v7.9.6.2` and the latest version of `production` (https://github.com/specify/specify7/commit/9b3aa0884571edc23d8a598e8013fdea36dbf0d3)

<img width="791" alt="image" src="https://github.com/user-attachments/assets/4dfbcd11-cc1d-4c98-a29b-428e225cc877" />

**Reply by ghost** [reply]
You have to remove the condition (lines 45 and 53) on the front end for the distinct checkbox in order to test it. These are the relevant lines

https://github.com/specify/specify7/blob/7ff9c53f839046c4d136891258da94bbccd2b723/specifyweb/frontend/js_src/lib/components/QueryBuilder/Toolbar.tsx#L45-L53

And yes, querying directly on the taxon tree and geography tree
